src/services/utilities.ts

```diff
@@ -3314,5 +3314,33 @@ namespace ts {
         };
     }
 
+    export function getProgramForAutoImport(source: AutoImportSource, program: Program, host: CacheableExportInfoMapHost | LanguageServiceHost): Program {
+        switch (source) {
+            case AutoImportSourceKind.Program:
+                return program;
+            case AutoImportSourceKind.PackageJson:
+                return Debug.checkDefined(host.getPackageJsonAutoImportProvider?.(), "Could not retrieve package.json AutoImportProvider");
+            default:

```

Comment:
`default` feels weird here - this seems more like a third, well-understood case, rather than a catch-all.

Reply:
It is, but there’s no way to match on it in a switch/case because the string is arbitrary. It could be an if/elseif/else but I don’t know if that’s really an improvement?

Reply:
I think I missed what value it would take on in that case - the path to the project?

Reply:
Yep.

Reply:
Part of me thinks it should have a prefix or something to avoid collisions, but I'm probably just being paranoid.